plugins/parcelport/mpi/parcelport_mpi.cpp

```diff
@@ -272,16 +272,15 @@ namespace hpx { namespace parcelset { namespace policies { namespace mpi
         {
             if (ec) {
                 // all errors during early parcel handling are fatal
-                try {
-                    HPX_THROW_EXCEPTION(network_error, "early_write_handler",
+                boost::exception_ptr exception =
+                    hpx::detail::get_exception(hpx::exception(ec),
+                        "mpi::early_write_handler", __FILE__, __LINE__,
                         "error while handling early parcel: " +
                             ec.message() + "(" +
-                            boost::lexical_cast<std::string>(ec.value())+ ")");
-                }
-                catch (hpx::exception const& e) {
-                    hpx::detail::report_exception_and_terminate(e);
-                }
-                return;
+                            boost::lexical_cast<std::string>(ec.value()) +
+                            ")" + parcelset::dump_parcel(p));
+
+                hpx::report_error(exception);
```

Comment:
Is the parcelport the only place where we have those constructs? I am pretty sure I just copy and pasted that from somewhere else.